### PR TITLE
Handle insecure TLS verification in TestUpAutocreateV2

### PR DESCRIPTION
**Pull request by AgustinRamiroDiaz**
# Proposed changes

For [TestUpAutocreateV2](https://github.com/okteto/okteto/blob/6951592d69ecc6a918595b0f2ca784052f51a641/integration/up/autocreate_test.go#L179):
- Feature to enable insecure TLS connections: can be enabled via the envvar `OKTETO_INSECURE_SKIP_TLS_VERIFY`
- Handle workspace dir in kubectl commands: set to run the `okteto kubetoken` command from the `kubeconfig auth section` with the `OKTETO_HOME` envvar

## Why

We have an e2e test in okteto/app that internally runs clones this repo and runs the `TestUpAutocreteV2` test. We need to run this tests against clusters without certificates, therefore this feature

### Considerations
- I didn't add this feature to all tests since it's not required. If you think this is a useful feature request, I can work on adding it to all tests
- I tried following the code patterns that I saw in the tests, I don't want to be too disruptive

## Testing done

I've ran the tests with the flag enabled
```
az@betsy ~/d/o/o/integration (agustin/handle-tls-envvar) [1]> OKTETO_INSECURE_SKIP_TLS_VERIFY=true OKTETO_URL=https://okteto.agustincluster.dev.okteto.net OKTETO_USER=okteto-admin OKTETO_PATH=/home/az/dev/okteto/okteto/bin/okteto OKTETO_APPS_SUBDOMAIN=agustincluster.dev.okteto.net go test github.com/okteto/okteto/integration/up -tags="integration" --count=1 -v -timeout 45m -run "TestUpAutocreateV2\$"
=== RUN   TestUpAutocreateV2
=== PAUSE TestUpAutocreateV2
=== CONT  TestUpAutocreateV2
2023/09/22 09:43:32 using /home/az/dev/okteto/okteto/bin/okteto
2023/09/22 09:43:33 okteto version 893c3ef7 

2023/09/22 09:43:37 creating namespace testupautocreatev2-linux-1695386613-okteto-admin
2023/09/22 09:43:37 Running: /home/az/dev/okteto/okteto/bin/okteto namespace create testupautocreatev2-linux-1695386613-okteto-admin -l debug
2023/09/22 09:43:40 create namespace output: 
time="2023-09-22T09:43:37-03:00" level=debug msg="server name override: \"\""
time="2023-09-22T09:43:37-03:00" level=info msg="started /home/az/dev/okteto/okteto/bin/okteto namespace create testupautocreatev2-linux-1695386613-okteto-admin -l debug"
time="2023-09-22T09:43:37-03:00" level=debug msg="certificate issuer CN=okteto-wildcard-ca"
time="2023-09-22T09:43:37-03:00" level=debug msg="context certificate not trusted by system roots: x509: certificate signed by unknown authority"
time="2023-09-22T09:43:39-03:00" level=info msg="Namespace 'testupautocreatev2-linux-1695386613-okteto-admin' created"
 ✓  Namespace 'testupautocreatev2-linux-1695386613-okteto-admin' created
time="2023-09-22T09:43:40-03:00" level=info msg="Using testupautocreatev2-linux-1695386613-okteto-admin @ okteto.agustincluster.dev.okteto.net as context"
 i  Using testupautocreatev2-linux-1695386613-okteto-admin @ okteto.agustincluster.dev.okteto.net as context
time="2023-09-22T09:43:40-03:00" level=info msg="failed to send analytics: analytics disabled by admin"
time="2023-09-22T09:43:40-03:00" level=info msg="finished /home/az/dev/okteto/okteto/bin/okteto namespace create testupautocreatev2-linux-1695386613-okteto-admin -l debug"

kubeconfig output:  ✓  Updated kubernetes context 'okteto_agustincluster_dev_okteto_net/testupautocreatev2-linux-1695386613-okteto-admin' in '[/tmp/TestUpAutocreateV23406161794/001/.kube/config]'
2023/09/22 09:43:47 original 'index.html' content: testupautocreatev2-linux-1695386613-okteto-admin
2023/09/22 09:43:47 Running up command: /home/az/dev/okteto/okteto/bin/okteto up -f /tmp/TestUpAutocreateV23406161794/001/okteto.yml -n testupautocreatev2-linux-1695386613-okteto-admin
2023/09/22 09:43:47 waiting for okteto up to be ready
2023/09/22 09:43:57 failed to read state file /tmp/TestUpAutocreateV23406161794/001/.okteto/testupautocreatev2-linux-1695386613-okteto-admin/autocreate/okteto.state: open /tmp/TestUpAutocreateV23406161794/001/.okteto/testupautocreatev2-linux-1695386613-okteto-admin/autocreate/okteto.state: no such file or directory
2023/09/22 09:44:07 okteto up is: attaching
2023/09/22 09:44:17 okteto up is: pulling
2023/09/22 09:44:26 okteto up is: ready
2023/09/22 09:44:29 waitForDeployment output: deployment "autocreate-okteto" successfully rolled out
2023/09/22 09:44:29 deployment "autocreate-okteto" successfully rolled out

2023/09/22 09:44:39 Running exec command: /home/az/dev/okteto/okteto/bin/okteto exec -n testupautocreatev2-linux-1695386613-okteto-admin -f /tmp/TestUpAutocreateV23406161794/001/okteto.yml -- cat .stignore | grep '(?d)venv'
2023/09/22 09:44:46 destroying pods with label selector: app=autocreate
2023/09/22 09:44:48 destroying pod autocreate-okteto-69959b4966-h6ps5
2023/09/22 09:44:54 endpoint http://localhost:8081/index.html didn't respond
2023/09/22 09:44:59 endpoint http://localhost:8081/index.html didn't respond
2023/09/22 09:45:09 okteto down output:
 i  Using testupautocreatev2-linux-1695386613-okteto-admin @ okteto.agustincluster.dev.okteto.net as context
 ✓  Development container 'autocreate' deactivated
 ✓  Persistent volume 'autocreate' removed
2023/09/22 09:45:09 okteto up exited: exit status 1.
Output:
 i  Using testupautocreatev2-linux-1695386613-okteto-admin @ okteto.agustincluster.dev.okteto.net as context
 i  Build section is not defined in your okteto manifest
 ✓  Client certificates generated
Installing dependencies...
syncthing-linux-amd64-v1.24.0.tar.gz  1003.81 KiB / 9.74 MiB [->______________] 10.06% ? p/ssyncthing-linux-amd64-v1.24.0.tar.gz  2.42 MiB / 9.74 MiB [---->______________] 24.82% ? p/ssyncthing-linux-amd64-v1.24.0.tar.gz  4.09 MiB / 9.74 MiB [------->___________] 41.98% ? p/ssyncthing-linux-amd64-v1.24.0.tar.gz  5.48 MiB / 9.74 MiB [------>_____] 56.25% 7.50 MiB p/ssyncthing-linux-amd64-v1.24.0.tar.gz  6.90 MiB / 9.74 MiB [-------->___] 70.85% 7.50 MiB p/ssyncthing-linux-amd64-v1.24.0.tar.gz  7.86 MiB / 9.74 MiB [--------->__] 80.63% 7.50 MiB p/ssyncthing-linux-amd64-v1.24.0.tar.gz  8.90 MiB / 9.74 MiB [---------->_] 91.37% 7.39 MiB p/ssyncthing-linux-amd64-v1.24.0.tar.gz  9.74 MiB / 9.74 MiB [-----------] 100.00% 7.23 MiB p/s ✓  Dependencies successfully installed
 ✓  Persistent volume successfully attached
 ✓  Images successfully pulled
 ✓  Files synchronized
    Context:   okteto.agustincluster.dev.okteto.net
    Namespace: testupautocreatev2-linux-1695386613-okteto-admin
    Name:      autocreate
    Forward:   8081 -> 8080

Serving HTTP on 0.0.0.0 port 8080 (http://0.0.0.0:8080/) ...
127.0.0.1 - - [22/Sep/2023 12:44:31] "GET /var.html HTTP/1.1" 200 -
127.0.0.1 - - [22/Sep/2023 12:44:33] "GET /index.html HTTP/1.1" 200 -
127.0.0.1 - - [22/Sep/2023 12:44:34] "GET /index.html HTTP/1.1" 200 -
127.0.0.1 - - [22/Sep/2023 12:44:35] "GET /index.html HTTP/1.1" 200 -
10.44.8.96 - - [22/Sep/2023 12:44:37] "GET /index.html HTTP/1.1" 200 -
127.0.0.1 - - [22/Sep/2023 12:44:39] "GET /index.html HTTP/1.1" 200 -
127.0.0.1 - - [22/Sep/2023 12:44:46] "GET /index.html HTTP/1.1" 200 -

Connection lost to your development container, reconnecting...
 ✓  Images successfully pulled
 ✓  Files synchronized
    Context:   okteto.agustincluster.dev.okteto.net
    Namespace: testupautocreatev2-linux-1695386613-okteto-admin
    Name:      autocreate
    Forward:   8081 -> 8080

Serving HTTP on 0.0.0.0 port 8080 (http://0.0.0.0:8080/) ...
127.0.0.1 - - [22/Sep/2023 12:45:04] "GET /index.html HTTP/1.1" 200 -

 x  Development container has been deactivated
    Find additional logs at: /tmp/TestUpAutocreateV23406161794/001/.okteto/testupautocreatev2-linux-1695386613-okteto-admin/autocreate/okteto.log
2023/09/22 09:45:10 okteto delete namespace testupautocreatev2-linux-1695386613-okteto-admin
--- PASS: TestUpAutocreateV2 (109.63s)
PASS
ok      github.com/okteto/okteto/integration/up 109.644s
```

## How to validate

You can run the integration tests against any cluster.

I've created a branch so that I trigger the tests with these changes. The [circleci job can be seen here](https://app.circleci.com/pipelines/github/okteto/okteto/11749/workflows/3d8e17ec-2d04-466c-8710-6108f607d4b2)

## CLI Quality Reminders 🔧

For both authors and reviewers:

- Scrutinize for potential regressions
- Ensure key automated tests are in place
- Build the CLI and test using the validation steps
- Assess Developer Experience impact (log messages, performances, etc)
- If too broad, consider breaking into smaller PRs
- Adhere to our [code style](https://github.com/okteto/okteto/blob/master/docs/code-style.md) and [code review](https://github.com/okteto/okteto/blob/master/docs/code-review.md) guidelines

### CODE REVIEW

File: integration/up/autocreate_test.go [hunk 1]
```diff
@@ -192,7 +204,14 @@ func TestUpAutocreateV2(t *testing.T) {
...
 	defer commands.RunOktetoDeleteNamespace(oktetoPath, namespaceOpts)
 	require.NoError(t, commands.RunOktetoKubeconfig(oktetoPath, dir))
-	c, _, err := okteto.NewK8sClientProvider().Provide(kubeconfig.Get([]string{filepath.Join(dir, ".kube", "config")}))
+	config := kubeconfig.Get([]string{filepath.Join(dir, ".kube", "config")})
+	for _, authInfo := range config.AuthInfos {
+		homeEnvVar := api.ExecEnvVar{Name: constants.OktetoHomeEnvVar, Value: dir}
+		authInfo.Exec.Env = append(authInfo.Exec.Env, homeEnvVar)
+	}
+
+	c, _, err := okteto.NewK8sClientProvider().Provide(config)
```
Comment (AgustinRamiroDiaz) [inline integration/up/autocreate_test.go:213]: 2 things to mention here:
a) I couldn't find a simpler way to do this. Please comment if you think of some
b) If we want to go ahead and add this to all tests, we should probably refactor to simplify writing tests and add this kind of logic to the `kubeconfig.Get func`

**Review by teresaromero — APPROVED** [review]
reviewed offline with agusting. Just a concern regarding the coupling of cli-app in terms of testing and integration.

if possible i would add this as patch to current beta in order to have a valid binary released to integrate with the app PR

cc @AdrianPedriza as you are handling the CLI release

**Comment by AgustinRamiroDiaz** [comment]
> if possible i would add this as patch to current beta in order to have a valid binary released to integrate with the app PR

Minor comment: there are alternative ways of doing it but I think this one is the cleanest

File: integration/up/autocreate_test.go [hunk 2]
```diff
@@ -192,7 +204,14 @@ func TestUpAutocreateV2(t *testing.T) {
 	require.NoError(t, commands.RunOktetoCreateNamespace(oktetoPath, namespaceOpts))
 	defer commands.RunOktetoDeleteNamespace(oktetoPath, namespaceOpts)
 	require.NoError(t, commands.RunOktetoKubeconfig(oktetoPath, dir))
-	c, _, err := okteto.NewK8sClientProvider().Provide(kubeconfig.Get([]string{filepath.Join(dir, ".kube", "config")}))
+	config := kubeconfig.Get([]string{filepath.Join(dir, ".kube", "config")})
+	for _, authInfo := range config.AuthInfos {
+		homeEnvVar := api.ExecEnvVar{Name: constants.OktetoHomeEnvVar, Value: dir}
+		authInfo.Exec.Env = append(authInfo.Exec.Env, homeEnvVar)
```
Comment (AdrianPedriza) [inline integration/up/autocreate_test.go:210]: in this to add okteto in the path right? should it not be the responsibility of the client instead the test?

**Comment by AdrianPedriza** [comment]
can we exec integration tests to check everything passes?

**Pull request closed without merging**